sgt-back-end/index.js

```diff
@@ -0,0 +1,130 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const jsonMiddleWare = express.json();
+
+app.use(jsonMiddleWare);
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select *
+    from grades
+    `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    })
+    .catch(error => {
+      // eslint-disable-next-line no-console
+      console.log(error);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const body = req.body;
+  const score = Number(req.body.score);
+  if (!body.name || !body.course || !body.score) {
+    res.status(400).json({ error: 'Entry must contain "Name", "Course", and "Score"' });
+    return;
+  } else if (!score || score < 0 || score > 100 || !Number.isInteger(score)) {
+    res.status(400).json({ error: 'Invalid integer' });
+    return;
+  }
+  const sql = `
+  insert into "grades" ("name", "course", "score")
+  values ($1, $2, $3)
+  returning *
+  `;
+  const values = [body.name, body.course, body.score];
+  db.query(sql, values)
+    .then(result => {
+      const grades = result.rows[0];
+      res.status(201).json(grades);
+    })
+    .catch(error => {
+      // eslint-disable-next-line no-console
+      console.log(error);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const body = req.body;
+  const id = Number(req.params.gradeId);
+  const score = Number(req.body.score);
+  if (!id) {
+    res.status(400).json({ error: 'Invalid Id' });
+    return;
```

Comment:
Fractional and negative `id`s should also trigger a `400 Bad Request`:

---

![image](https://user-images.githubusercontent.com/16143771/151639289-e4e65934-7ab9-4fff-8038-cfa1d04121e5.png)